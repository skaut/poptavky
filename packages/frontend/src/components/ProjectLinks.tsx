import { ReactNode } from "react"
import { AiFillGithub } from "react-icons/ai"
import { MdWebAsset } from "react-icons/md"
import { GrMail } from "react-icons/gr"
import { BiBookAlt } from "react-icons/bi"
import { FaSlack, FaFacebook } from "react-icons/fa"
import { MdChecklist } from "react-icons/md"
import { ImEarth } from "react-icons/im"
import { HiOutlineDocumentText } from "react-icons/hi"
import { BsFillPersonFill } from "react-icons/bs"
import { ExtLink } from "./ExtLink"
import { Paragraph, Mark, SmallLink } from "./Typography"
import { ProjectInfo, ProjectInfoLink } from "../interfaces/ProjectInfo"

export interface LinkType {
  type: ProjectInfoLink["type"]
  label: string
  icon?: ReactNode
}

export const links: LinkType[] = [
  {
    type: "email",
    label: "E-mail",
    icon: <GrMail />,
  },
  {
    type: "demo",
    label: "Demo",
    icon: <MdWebAsset />,
  },
  {
    type: "docs",
    label: "Dokumentace",
    icon: <HiOutlineDocumentText />,
  },
  {
    type: "facebook-group",
    label: "Skupina na facebooku",
    icon: <FaFacebook />,
  },
  {
    type: "facebook-page",
    label: "Stránka na facebooku",
    icon: <FaFacebook />,
  },
  {
    type: "github-repo",
    label: "Repozitář",
    icon: <AiFillGithub />,
  },
  {
    type: "homepage",
    label: "Web",
    icon: <ImEarth />,
  },
  {
    type: "issue-tracker",
    label: "Issue tracker",
    icon: <MdChecklist />,
  },
  {
    type: "slack",
    label: "Slack",
    icon: <FaSlack />,
  },
  {
    type: "wiki",
    label: "Wiki",
    icon: <BiBookAlt />,
  },
]

export const ProjectLinks = ({ projectInfo }: { projectInfo: ProjectInfo }) => (
  <>
    <Paragraph>
      <Mark>
        <BsFillPersonFill />
        &nbsp;
        {projectInfo.maintainers.length > 1 ? "Správci:" : "Správce:"}
      </Mark>
      {projectInfo.maintainers
        .filter(
          (person): person is { name: string; email: string } =>
            person.email !== undefined
        )
        .map((person) => (
          <SmallLink key={person.email}>
            <ExtLink href={`mailto:${person.email}`}>{person.name}</ExtLink>
          </SmallLink>
        ))}
    </Paragraph>
    {links.map((link) => {
      const currentLink = projectInfo.links.find(
        (item) => item.type === link.type
      )
      if (!currentLink) {
        return undefined
      }
      const url = new URL(currentLink.uri)
      return (
        <Paragraph key={link.type}>
          <Mark>
            {link.icon}&nbsp;{link.label}:
          </Mark>
          <SmallLink key={link.type}>
            <ExtLink href={currentLink.uri}>
              {`${url.host}${url.pathname}`}
            </ExtLink>
          </SmallLink>
        </Paragraph>
      )
    })}
  </>
)
