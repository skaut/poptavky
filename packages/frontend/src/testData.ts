import { ProjectListings } from "./interfaces/ProjectListings"

export const testData: ProjectListings = {
  projects: [
    {
      owner: "skaut",
      repo: "skaut-google-drive-gallery",
      info: {
        name: "WordPress plugin: Galerie obrázků a videí z Google Disku",
        "short-description": "Plugin pro WordPress, který ze složek s fotkami na Google Disku dělá galerie na webu.",
        description: "Galerie obrázků a videí z Google Disku je plugin pro WordPress, který propojí web s Google Diskem. V pluginu se dá vybrat libovolná složku na Google Disku nebo ve Sdíleném disku a plugin z ní vytvoří galerii s obrázky a videi na libovolné stránce či příspěvku na webu. Ze všech složek a podsložek jsou navíc vytvořeny vnořené “podgalerie”. Plugin vzniknul primárně pro potřeby Junáka, oddílů a středisek, ale vyvíjíme ho tzv. Open-Source (otevřeně pro veřejnost) a nabízíme ho v oficiální databázi pluginů pro WordPress. V tuto chvíli (léto 2021) podle statistik WordPressu plugin běží na víc jak 4 000 webových stránek a z kontaktů s uživateli víme, že ho využívají jak oddíly a střediska v Junáku, tak i \"náhodná veřejnost\" - zvlášť oblíbený se zdá být u skautů z celého světa a profesionálních fotografů, prezentujících svoje fotky.",
        tags: [
          "wordpress",
          "php",
          "javascript",
          "google"
        ],
        maintainers: [
          {
            name: "Marek Dědič – Mlha",
            email: "mlha@skaut.cz"
          }
        ],
        links: [
          {
            type: "email",
            uri: "mailto:mlha@skaut.cz"
          },
          {
            type: "homepage",
            uri: "https://wordpress.org/plugins/skaut-google-drive-gallery/"
          },
          {
            type: "demo",
            uri: "https://demo-skaut-google-drive-gallery.skauting.cz/"
          },
          {
            type: "issue-tracker",
            uri: "https://github.com/skaut/skaut-google-drive-gallery/issues"
          },
          {
            type: "docs",
            uri: "https://napoveda.skaut.cz/dobryweb/cs-skaut-google-drive-gallery"
          },
          {
            type: "github-repo",
            uri: "https://github.com/skaut/skaut-google-drive-gallery",
            name: "skaut/skaut-google-drive-gallery"
          }
        ]
      },
      issues: [
        {
          number: 1,
          title: "Add hint to set videos to public",
          description: "When editing and selecting a folder with videos, the editor could display a notice that setting videos to \"Anyone can view\" will speed them up.\r\n\r\nDepends on #792",
          link: "https://github.com/skaut/skaut-google-drive-gallery/issues/873"
        },
        {
          number: 2,
          title: "If a shortcut name differs from the original, the name in gallery and breadcrumbs differs",
          description: "When in a situation that there is a Google Drive shortcut somewhere in the gallery, the plugin displays the folder and it works for browsing. However, when looking at the shortcut as a member of its parent directory and when being inside the directory and looking at its name in the breadcrumbs, one uses the shortcut name, whereas the other uses the original folder name (when they differ).\r\n\r\nThis is a bug and both should show the shortcut name.",
          link: "https://github.com/skaut/skaut-google-drive-gallery/issues/745"
        },
        {
          number: 3,
          title: "Editing doesn't check path validity",
          description: "When opening the editor which contains a gallery that is made from a folder that has been renamed or deleted, the editor breaks. It shows: `Directory &quot;Galerie ODYWEB&quot; wasn&#039;t found - it may have been deleted or renamed.` Instead, it should verify that the path is valid and if it's not, should display a warning (and possibly move you to either the root directory, or the closest valid ancestor - however, this should not lead to you accidentally changing the gallery!)\r\n\r\nLast tested on 36dc1a10a353ba5bf2b98f33b49b1135a36ae3d3",
          link: "https://github.com/skaut/skaut-google-drive-gallery/issues/683"
        },
        {
          number: 4,
          title: "Ctrl-click on folders does not work",
          description: "Ctrl+click on a gallery item (subfolder) or breadcrumbs should open in a new tab",
          link: "https://github.com/skaut/skaut-google-drive-gallery/issues/571"
        },
        {
          number: 5,
          title: "Add option to specify gallery by a Drive link",
          description: "https://wordpress.org/support/topic/folder-link-instead-of-folder-name/\r\n\r\nBasically, instead of searching for the folder in WP, you'd just paste the link to it into the editor and the plugin would figure out the path (Google Drive files now return a list of parents, so it could be done.)\r\n\r\nDepends on #111.",
          link: "https://github.com/skaut/skaut-google-drive-gallery/issues/311"
        },
        {
          number: 6,
          title: "Add images to the block",
          description: "Currently, images aren't listed in the editor - both block and TinyMCE plugin. This often confuses users.",
          link: "https://github.com/skaut/skaut-google-drive-gallery/issues/257"
        },
        {
          number: 7,
          title: "Add video overlay",
          description: "In the image grid, we should add an overlay to signify that an item is a video (for example a big play icon).",
          link: "https://github.com/skaut/skaut-google-drive-gallery/issues/235"
        },
        {
          number: 8,
          title: "Increase test coverage",
          description: "https://codecov.io/gh/skaut/skaut-google-drive-gallery",
          link: "https://github.com/skaut/skaut-google-drive-gallery/issues/164"
        },
        {
          number: 9,
          title: "Themes with \"editor-styles\" support change the look of the block editor",
          description: "If the active theme contains\r\n```php\r\nadd_theme_support( 'editor-styles' );\r\n```\r\nits styles are also applied to the block editor, which then looks weird. We should implement our own styles for the editor.\r\n\r\nExample of such theme is the Twenty Seventeen theme.",
          link: "https://github.com/skaut/skaut-google-drive-gallery/issues/98"
        },
        {
          number: 10,
          title: "Google app creation manual is outdated",
          description: "https://napoveda.skaut.cz/dobryweb/en-skaut-google-drive-gallery/en-get-google-application\r\nhttps://napoveda.skaut.cz/dobryweb/cs-skaut-google-drive-gallery/cs-ziskani-google-aplikace\r\n\r\nGoogle changed the consent screen layout and require some form of verification.",
          link: "https://github.com/skaut/skaut-google-drive-gallery/issues/92"
        },
        {
          number: 11,
          title: "Directory preview images",
          description: "Currently, the first image in a directory is used for its preview.",
          link: "https://github.com/skaut/skaut-google-drive-gallery/issues/54"
        },
        {
          number: 12,
          title: "UX & design audit",
          description: "Go over the design of:\r\n- [ ] gallery view https://github.com/skaut/skaut-google-drive-gallery/issues/36\r\n- [ ] preview\r\n- [ ] folder thumbnail - overlay, folder title and image count\r\n- [ ] svg folder icon\r\n- [ ] gallery breadcrumbs\r\n- [ ] options page\r\n- [ ] WP.org plugin page\r\n- [ ] TinyMCE & Gutenberg editor plugins\r\n- [ ] TinyMCE & Gutenberg \"Authentication failed\" dialogs\r\n\r\nEverything on the frontend needs to work at least with the default WP template and DSW templates.",
          link: "https://github.com/skaut/skaut-google-drive-gallery/issues/33"
        }
      ]
    },
    {
      owner: "skaut",
      repo: "skautis-integration",
      info: {
        name: "WordPress plugin: SkautIS integrace",
        "short-description": "Plugin pro WordPress, který napojuje přihlašování přes skautIS na skautské weby.",
        description: "Plugin zajišťuje propojení skautských webů se skautISem. Dá se přes plugin registrovat, přihlašovat. Dokáže u stránek nastavovat oprávnění podle roli a dalších nastavení ve skautISu. Skrývá obsah podle toho jaká pravidla na stránce jsou nastavená.",
        tags: [
          "wordpress",
          "php",
          "javascript",
          "skautis"
        ],
        maintainers: [
          {
            name: "Michal Janata – Kalich",
            email: "kalich@skaut.cz"
          },
          {
            name: "Marek Dědič – Mlha",
            email: "mlha@skaut.cz"
          }
        ],
        links: [
          {
            type: "email",
            uri: "mailto:kalich@skaut.cz"
          },
          {
            type: "homepage",
            uri: "https://wordpress.org/plugins/skautis-integration/"
          },
          {
            type: "issue-tracker",
            uri: "https://github.com/skaut/skautis-integration/issues"
          },
          {
            type: "docs",
            uri: "https://napoveda.skaut.cz/skautis/skautis-integration"
          },
          {
            type: "github-repo",
            uri: "https://github.com/skaut/skautis-integration",
            name: "skaut/skautis-integration"
          }
        ]
      },
      issues: [
        {
          number: 1,
          title: "User management smart table is not smart when the registration module is enabled",
          description: "",
          link: "https://github.com/skaut/skautis-integration/issues/115"
        },
        {
          number: 2,
          title: "Chyba v zobrazení",
          description: "![image](https://user-images.githubusercontent.com/8988898/130583907-9aaf6288-2129-42dd-88ef-0a27b0c7ddd1.png)\r\n",
          link: "https://github.com/skaut/skautis-integration/issues/109"
        },
        {
          number: 3,
          title: "WordPress 5.x + Gutenberg",
          description: "Musíme pořešit, jak nejlépe integrovat náš plugin s WP 5.x a hlavně Gutenbergem.\r\n\r\nSkrývání bloků, celých stránek... atd",
          link: "https://github.com/skaut/skautis-integration/issues/70"
        },
        {
          number: 4,
          title: "Profilové fotky",
          description: "Co do WordPressu tahat i profilové fotky, co si lidi do skautISu přidávají. Aby u profilu nějakou fotku měli...",
          link: "https://github.com/skaut/skautis-integration/issues/43"
        },
        {
          number: 5,
          title: "Výběry pravidel v shortcode",
          description: "Přidat Select2 do výběru pravidel v shortcode.",
          link: "https://github.com/skaut/skautis-integration/issues/39"
        },
        {
          number: 6,
          title: "Modul \"Nábory\"",
          description: "modul pro nábory se zobrazením náborové mapy\r\npříklad: http://is.skaut.cz/napoveda/programatori.Hotove-reseni-MapAdvertising-Interaktivni-mapa-naborovych-informaci-zakladni.ashx\r\n\r\nOtevírám k tomu diskuzi...",
          link: "https://github.com/skaut/skautis-integration/issues/25"
        }
      ]
    }
  ]
}