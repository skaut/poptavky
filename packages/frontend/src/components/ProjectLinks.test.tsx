import renderer from "react-test-renderer"
import { ProjectLinks } from "./ProjectLinks"
import { testData } from "../testData"
import { MemoryRouter } from "react-router-dom"

const project = testData.projects[0]

describe("ProjectLinks component", () => {
  it("should renders correctly", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <ProjectLinks projectInfo={project.info} />
        </MemoryRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
