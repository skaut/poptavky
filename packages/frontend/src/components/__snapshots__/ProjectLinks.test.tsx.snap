// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectLinks component should render correctly 1`] = `
Array [
  .emotion-0 {
  margin: 0 0 16px;
}

.emotion-0 img {
  max-width: 100%;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #999;
  font-size: 0.8em;
}

.emotion-1 svg {
  margin-right: 0.2em;
}

.emotion-2 a {
  display: block;
  font-size: 0.9em;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.emotion-3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<p
    className="emotion-0"
  >
    <span
      className="emotion-1"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3 14s-1 0-1-1 1-4 6-4 6 3 6 4-1 1-1 1H3Zm5-6a3 3 0 1 0 0-6 3 3 0 0 0 0 6Z"
        />
      </svg>
       
      Správce:
    </span>
    <span
      className="emotion-2"
    >
      <a
        className="emotion-3"
        href="mailto:mlha@skaut.cz"
        rel="noreferrer noopener"
        target="_blank"
      >
        Marek Dědič – Mlha
      </a>
    </span>
  </p>,
  .emotion-0 {
  margin: 0 0 16px;
}

.emotion-0 img {
  max-width: 100%;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #999;
  font-size: 0.8em;
}

.emotion-1 svg {
  margin-right: 0.2em;
}

.emotion-2 a {
  display: block;
  font-size: 0.9em;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.emotion-3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<p
    className="emotion-0"
  >
    <span
      className="emotion-1"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M23,20 L23,6 L12,15 L1,6 L1,20 L23,20 Z M12,12 L22,4 L2,4 L12,12 Z"
          fillRule="evenodd"
        />
      </svg>
       
      E-mail
      :
    </span>
    <span
      className="emotion-2"
    >
      <a
        className="emotion-3"
        href="mailto:mlha@skaut.cz"
        rel="noreferrer noopener"
        target="_blank"
      >
        mlha@skaut.cz
      </a>
    </span>
  </p>,
  .emotion-0 {
  margin: 0 0 16px;
}

.emotion-0 img {
  max-width: 100%;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #999;
  font-size: 0.8em;
}

.emotion-1 svg {
  margin-right: 0.2em;
}

.emotion-2 a {
  display: block;
  font-size: 0.9em;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.emotion-3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<p
    className="emotion-0"
  >
    <span
      className="emotion-1"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M19 4H5a2 2 0 00-2 2v12a2 2 0 002 2h14c1.1 0 2-.9 2-2V6a2 2 0 00-2-2zm0 14H5V8h14v10z"
        />
      </svg>
       
      Demo
      :
    </span>
    <span
      className="emotion-2"
    >
      <a
        className="emotion-3"
        href="https://demo-skaut-google-drive-gallery.skauting.cz/"
        rel="noreferrer noopener"
        target="_blank"
      >
        demo-skaut-google-drive-gallery.skauting.cz/
      </a>
    </span>
  </p>,
  .emotion-0 {
  margin: 0 0 16px;
}

.emotion-0 img {
  max-width: 100%;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #999;
  font-size: 0.8em;
}

.emotion-1 svg {
  margin-right: 0.2em;
}

.emotion-2 a {
  display: block;
  font-size: 0.9em;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.emotion-3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<p
    className="emotion-0"
  >
    <span
      className="emotion-1"
    >
      <svg
        aria-hidden="true"
        fill="none"
        height="1em"
        stroke="currentColor"
        strokeWidth="2"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9 12h6m-6 4h6m2 5H7a2 2 0 01-2-2V5a2 2 0 012-2h5.586a1 1 0 01.707.293l5.414 5.414a1 1 0 01.293.707V19a2 2 0 01-2 2z"
          strokeLinecap="round"
          strokeLinejoin="round"
        />
      </svg>
       
      Dokumentace
      :
    </span>
    <span
      className="emotion-2"
    >
      <a
        className="emotion-3"
        href="https://napoveda.skaut.cz/dobryweb/cs-skaut-google-drive-gallery"
        rel="noreferrer noopener"
        target="_blank"
      >
        napoveda.skaut.cz/dobryweb/cs-skaut-google-drive-gallery
      </a>
    </span>
  </p>,
  .emotion-0 {
  margin: 0 0 16px;
}

.emotion-0 img {
  max-width: 100%;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #999;
  font-size: 0.8em;
}

.emotion-1 svg {
  margin-right: 0.2em;
}

.emotion-2 a {
  display: block;
  font-size: 0.9em;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.emotion-3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<p
    className="emotion-0"
  >
    <span
      className="emotion-1"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M511.6 76.3C264.3 76.2 64 276.4 64 523.5 64 718.9 189.3 885 363.8 946c23.5 5.9 19.9-10.8 19.9-22.2v-77.5c-135.7 15.9-141.2-73.9-150.3-88.9C215 726 171.5 718 184.5 703c30.9-15.9 62.4 4 98.9 57.9 26.4 39.1 77.9 32.5 104 26 5.7-23.5 17.9-44.5 34.7-60.8-140.6-25.2-199.2-111-199.2-213 0-49.5 16.3-95 48.3-131.7-20.4-60.5 1.9-112.3 4.9-120 58.1-5.2 118.5 41.6 123.2 45.3 33-8.9 70.7-13.6 112.9-13.6 42.4 0 80.2 4.9 113.5 13.9 11.3-8.6 67.3-48.8 121.3-43.9 2.9 7.7 24.7 58.3 5.5 118 32.4 36.8 48.9 82.7 48.9 132.3 0 102.2-59 188.1-200 212.9a127.5 127.5 0 0 1 38.1 91v112.5c.8 9 0 17.9 15 17.9 177.1-59.7 304.6-227 304.6-424.1 0-247.2-200.4-447.3-447.5-447.3z"
        />
      </svg>
       
      Repozitář
      :
    </span>
    <span
      className="emotion-2"
    >
      <a
        className="emotion-3"
        href="https://github.com/skaut/skaut-google-drive-gallery"
        rel="noreferrer noopener"
        target="_blank"
      >
        github.com/skaut/skaut-google-drive-gallery
      </a>
    </span>
  </p>,
  .emotion-0 {
  margin: 0 0 16px;
}

.emotion-0 img {
  max-width: 100%;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #999;
  font-size: 0.8em;
}

.emotion-1 svg {
  margin-right: 0.2em;
}

.emotion-2 a {
  display: block;
  font-size: 0.9em;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.emotion-3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<p
    className="emotion-0"
  >
    <span
      className="emotion-1"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        version="1.1"
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 0c-4.418 0-8 3.582-8 8s3.582 8 8 8 8-3.582 8-8-3.582-8-8-8zM8 15c-0.984 0-1.92-0.203-2.769-0.57l3.643-4.098c0.081-0.092 0.126-0.21 0.126-0.332v-1.5c0-0.276-0.224-0.5-0.5-0.5-1.765 0-3.628-1.835-3.646-1.854-0.094-0.094-0.221-0.146-0.354-0.146h-2c-0.276 0-0.5 0.224-0.5 0.5v3c0 0.189 0.107 0.363 0.276 0.447l1.724 0.862v2.936c-1.813-1.265-3-3.366-3-5.745 0-1.074 0.242-2.091 0.674-3h1.826c0.133 0 0.26-0.053 0.354-0.146l2-2c0.094-0.094 0.146-0.221 0.146-0.354v-1.21c0.634-0.189 1.305-0.29 2-0.29 1.1 0 2.141 0.254 3.067 0.706-0.065 0.055-0.128 0.112-0.188 0.172-0.567 0.567-0.879 1.32-0.879 2.121s0.312 1.555 0.879 2.121c0.569 0.569 1.332 0.879 2.119 0.879 0.049 0 0.099-0.001 0.149-0.004 0.216 0.809 0.605 2.917-0.131 5.818-0.007 0.027-0.011 0.055-0.013 0.082-1.271 1.298-3.042 2.104-5.002 2.104z"
        />
      </svg>
       
      Web
      :
    </span>
    <span
      className="emotion-2"
    >
      <a
        className="emotion-3"
        href="https://wordpress.org/plugins/skaut-google-drive-gallery/"
        rel="noreferrer noopener"
        target="_blank"
      >
        wordpress.org/plugins/skaut-google-drive-gallery/
      </a>
    </span>
  </p>,
  .emotion-0 {
  margin: 0 0 16px;
}

.emotion-0 img {
  max-width: 100%;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #999;
  font-size: 0.8em;
}

.emotion-1 svg {
  margin-right: 0.2em;
}

.emotion-2 a {
  display: block;
  font-size: 0.9em;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.emotion-3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<p
    className="emotion-0"
  >
    <span
      className="emotion-1"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M22 7h-9v2h9V7zm0 8h-9v2h9v-2zM5.54 11L2 7.46l1.41-1.41 2.12 2.12 4.24-4.24 1.41 1.41L5.54 11zm0 8L2 15.46l1.41-1.41 2.12 2.12 4.24-4.24 1.41 1.41L5.54 19z"
        />
      </svg>
       
      Issue tracker
      :
    </span>
    <span
      className="emotion-2"
    >
      <a
        className="emotion-3"
        href="https://github.com/skaut/skaut-google-drive-gallery/issues"
        rel="noreferrer noopener"
        target="_blank"
      >
        github.com/skaut/skaut-google-drive-gallery/issues
      </a>
    </span>
  </p>,
]
`;
